semgrep/semgrep/rule_lang.py

```diff
@@ -258,12 +291,26 @@ def parse_yaml_preserve_spans(contents: str, filename: Optional[str]) -> YamlTre
     parse yaml into a YamlTree object. The resulting spans are tracked in SourceTracker
     so they can be used later when constructing error messages or displaying context.
     """
-    # this uses the `RoundTripConstructor` which inherits from `SafeConstructor`
     source_hash = SourceTracker.add_source(contents)
 
+    # this uses the `RoundTripConstructor` which inherits from `SafeConstructor`
     class SpanPreservingRuamelConstructor(RoundTripConstructor):
         def construct_object(self, node: Node, deep: bool = False) -> YamlTree:
             r = super().construct_object(node, deep)
+            if r is None:
+                from semgrep.error import InvalidRuleSchemaError
```

Comment:
I would move this to the top of the file. It's nice to have all your imports in one place at the top vs. sprinkled throughout.

Reply:
Circular imporr